File: _backup/StatusPageImporter/EsUtil.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

using Elasticsearch.Net;
using Elasticsearch.Net.Connection;
using Newtonsoft.Json.Linq;

namespace StatusPageImporter
{
	static class EsUtil
	{
		public static string GetIndexName(DateTime day)
		{
			var indexName = "logstash-" + day.ToString("yyyy.MM.dd");
			return indexName;
		}

		public static List<IncidentRecord> GetRecords(string indexNames)
		{
			var records = new List<IncidentRecord>();

			var settings = new ConnectionConfiguration(new Uri(AppSettings.EsServerUrl));
			var client = new ElasticsearchClient(settings);
			var requestTemplate = File.ReadAllText("request.txt");

			var curSegment = 0;
			while (true)
			{
				var segment = curSegment;

				var request = requestTemplate.Replace("{0}", segment.ToString());
				var result = client.Search(indexNames, request);
				var hits = result.Response["hits"]["hits"].Value;

				foreach (var cur in hits)
				{
					var hit = (JObject) JObject.Parse(cur.ToString());
					var fields = hit["_source"];

					var timeToken = fields["@timestamp"];

					var record = new IncidentRecord
						{
							Time = timeToken.Value<DateTime>(),
							Message = fields["message_data"].ToString(),
						};
					records.Add(record);
				}

				if (hits.Count < SegmentSize)
					break;

				curSegment++;
			}

			records.Sort((x, y) => x.Time.CompareTo(y.Time));
			return records;
		}

		private const int SegmentSize = 1024;
	}
}

File: src/main/scala/EsUtil.scala
import java.net.HttpURLConnection

import org.joda.time.{DateTimeZone, DateTime}
import play.libs.Json

import scala.collection.JavaConversions._
import scala.collection.mutable.ArrayBuffer
import scalaj.http.{HttpOptions, Http}

object EsUtil {
  def getIndexName(day: DateTime): String =
  {
    val indexName = "logstash-" + day.withZone(DateTimeZone.UTC).toString("yyyy.MM.dd")
    return indexName
  }

  def getRecords(indexNames: String): Array[IncidentRecord] =
  {
    def requestFile = this.getClass().getResource("request.txt").getFile()
    val source = scala.io.Source.fromFile(requestFile)
    val requestBase = source.mkString
    source.close()

    var url = "http://api.cityindex.logsearch.io/%s/_search".format(indexNames)

    val records = ArrayBuffer[IncidentRecord]()

    var curSegment = 0
    while (true)
    {
      val segment = curSegment

      val requestText = requestBase.format(segment)

      val result = Http.postData(url, requestText)
        .header("Content-Type", "application/json")
        .header("Charset", "UTF-8")
        .option(HttpOptions.connTimeout(60 * 1000))
        .option(HttpOptions.readTimeout(60 * 1000))

      if (result.responseCode != HttpURLConnection.HTTP_OK)
        throw new Exception()

      val responseText = result.asString

      val json = Json.parse(responseText)
      val hits = json.at("/hits/hits")

      for (hit <- hits)
      {
        println(hit.toString())
        val sourceData = hit.at("/_source")
        val timeText = sourceData.at("/@timestamp").asText()
        var record = new IncidentRecord {
          time = DateTime.parse(timeText)
          message = sourceData.toString()
        }
        records += record
      }

      if (hits.size() < SegmentSize) {
        val sorted = records.sortWith((x, y) => x.time.isBefore(y.time))
        return sorted.toArray
      }

      curSegment += 1
    }

    throw new InternalError()
  }

  val SegmentSize = 1024
}
